src/amo/components/Search/SearchForm.js

```diff
@@ -0,0 +1,67 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { loadEntities } from 'core/actions';
+import { fetchAddon } from 'core/api';
+import { gettext as _ } from 'core/utils';
+
+import 'amo/css/SearchForm.scss';
+import 'amo/css/lib/buttons.scss';
+
+export class SearchFormBase extends React.Component {
+  static propTypes = {
+    pathname: PropTypes.string.isRequired,
+    query: PropTypes.string,
+  }
+  static contextTypes = {
+    router: PropTypes.object,
+  }
+
+  goToSearch(query) {
+    const { pathname } = this.props;
+    this.context.router.push(`${pathname}?q=${query}`);
+  }
+
+  handleSearch = () => {
+    this.goToSearch(this.refs.query.value);
+  }
+
+  handleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      this.handleSearch();
+    }
+  }
+
+  render() {
+    const { query } = this.props;
+    return (
+      <form ref="form" className="search-form" onKeyDown={this.handleKeyDown}
+            onSubmit={(e) => e.preventDefault()}>
+        <label className="visually-hidden" htmlFor="q">{_('Search')}</label>
+        <input ref="query" type="search" name="q" placeholder={_('Search')} defaultValue={query} />
+        <button className="button button-middle" ref="submit" type="submit" title="Enter"
```

Comment:
Instead of specifying a string for `ref` the new way is to pass a function which gets a reference to the element and that can be set as a property on the component.

```jsx
<button className="button button-middle"
        ref={(ref) => { this.submitButton = ref; }}>
...
</button>